packages/sommelier.rb

```diff
@@ -293,7 +344,29 @@ def self.build
           }
           if ! checksommelierwayland || ! checksommelierxwayland ; then
             [ -f  #{CREW_PREFIX}/bin/stopbroadway ] && stopbroadway
+            set -a
+            # Set default SCALE to 1.
+            SCALE=${SCALE:-1}
+            # via https://stackoverflow.com/questions/47027323/round-to-the-nearest-0-5-decimal-in-bash/47027557#47027557
+            function roundhalves {
+                    echo "$1 * 2" | bc | xargs -I@ printf "%1.f" @ | xargs -I% echo "% * .5" | bc
+            }
+            pxwidth=$(WAYLAND_DISPLAY=wayland-0 wayland-info -i wl_output | grep width: | grep px | head -n 1 | awk '{print $2}')
+            lwidth=$(WAYLAND_DISPLAY=wayland-0 wayland-info -i zxdg_output_manager_v1 | grep logical_width:  | sed 's/,//' | awk '{print $2}')
+            # echo "pxwidth: $pxwidth, lwidth: $lwidth"
+            # SCALE needs to be rounded to the nearest 0.5
+            # Check to see if pxwidth and lwidth are integers before calculating SCALE.
+            # wayland-info on armv7l does not show lwidth, but aarch64 does.
+            if [[ $pxwidth == ?(-)+([0-9]) ]] && [[ $lwidth == ?(-)+([0-9]) ]]; then
+              SCALE=$(roundhalves $(echo "scale=2 ;$lwidth / $pxwidth" | bc))
+            fi
+            # Allow overriding environment variables before starting sommelier daemon.
+            [ -f "$HOME/.sommelier.env" ] && source ~/.sommelier.env 2>> /usr/local/var/log/sommelier.log
+            set +a
+            echo -e "\e[1;33m""Sommelier SCALE is set to \e[1;32m"${SCALE}"\e[1;33m"."\e[0m"
+            echo -e "\e[1;33m""SCALE may be manually set in ~/.sommelier.env .""\e[0m"
             #{CREW_PREFIX}/sbin/sommelierd &>/dev/null &
```

Comment:
Replace `&> /dev/null` with `2>> #{CREW_PREFIX}/var/log/sommelier.log`?

Reply:
Sure, but let's dump all output in there with `&>> #{CREW_PREFIX}/var/log/sommelier.log`?

Reply:
(Keep in mind that is being backgrounded.)